tests/pfquickcheck/pflang.lua

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env luajit
+-- -*- lua -*-
+-- This module generates (a subset of) pflang, libpcap's filter language
+
+-- Convention: initial uppercase letter => generates pflang expression
+-- initial lowercase letter => aux helper
+
+module(..., package.seeall)
+local choose = require("pf.utils").choose
+local set = require("pf.utils").set
+
+local function Empty() return { "" } end
+
+-- Given something like { 'host', '127.0.0.1' }, make it sometimes
+-- start with src or dst. This should only be called on expressions
+-- which can start with src or dst!
+local function srcdstify(expr)
+   local r = math.random()
+   if r < 1/3 then table.insert(expr, 1, "src")
+   elseif r < 2/3 then table.insert(expr, 1, "dst")
+   end -- else: leave it unchanged
+   return expr
+end
+
+local function IPProtocol()
+   return choose({ "icmp", "igmp", "igrp", "pim", "ah", "esp", "vrrp",
+                    "udp", "tcp", "sctp", "icmp6" })
+end
+
+-- These are parsed seperately from IPProtocol values in pflua; does it make
+-- sense to mainain this separation?
```

Comment:
From the generator?  I don't know.  I guess not?  One TODO before merging would be to decide one way or the other and then adapt the code accordingly.